adapters/sharethrough/sharethrough.go

```
@@ -1,62 +1,103 @@
+	"net/http"
+	"strings"
 )
 
-const supplyId = "FGMrCMMc"
-const strVersion = 8
+var strVersion = "10.0"
+
+type SharethroughAdapter struct {
+	endpoint string
+}
 
-// Builder builds a new instance of the Sharethrough adapter for the given bidder with the given config.
 func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
 	bidder := &SharethroughAdapter{
-		AdServer: StrOpenRTBTranslator{
-			UriHelper: StrUriHelper{BaseURI: config.Endpoint, Clock: Clock{}},
-			Util:      Util{Clock: Clock{}},
-			UserAgentParsers: UserAgentParsers{
-				ChromeVersion:    regexp.MustCompile(`Chrome\/(?P<ChromeVersion>\d+)`),
-				ChromeiOSVersion: regexp.MustCompile(`CriOS\/(?P<chromeiOSVersion>\d+)`),
-				SafariVersion:    regexp.MustCompile(`Version\/(?P<safariVersion>\d+)`),
-			},
-		},
+		endpoint: config.Endpoint,
 	}
 	return bidder, nil
 }
 
-type SharethroughAdapter struct {
-	AdServer StrOpenRTBInterface
-}
-
 func (a SharethroughAdapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
-	var reqs []*adapters.RequestData
+	var requests []*adapters.RequestData
+	var errors []error
 
-	if request.Site == nil {
-		return nil, []error{fmt.Errorf("request must include a site; in-app placements are not supported")}
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	modifiableSource := openrtb2.Source{}
+	if request.Source != nil {
+		modifiableSource = *request.Source
+	}
+	var sourceExt map[string]interface{}
+	if err := json.Unmarshal(modifiableSource.Ext, &sourceExt); err == nil {
+		sourceExt["str"] = strVersion
+		sourceExt["version"] = version.Ver
+	} else {
+		sourceExt = map[string]interface{}{"str": strVersion, "version": version.Ver}
 	}
-	var domain = Util{}.parseDomain(request.Site.Page)
+	modifiableSource.Ext, _ = json.Marshal(sourceExt)
```

Eeek! Please do not ignore errors from `json.Marshal`.